README
====================
Antiviral
====================

Antiviral is a frontend for clamav

For more information, please visit our website:
http://launchpad.net/antiviral/
or
http://www.atareao.es

============
Requirements
============

Required dependencies:

	gir1.2-gtk-3.0,
	gir1.2-gdkpixbuf-2.0,
	gir1.2-notify-0.7

========
Download
========

Download the package:

    $ bzr branch lp:antiviral


debian/changelog
define-me (0.4.0-0extras13.04.1) raring; urgency=low

  * Changed configuration engine
  * Updated searches in RAE

 -- Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>  Sun, 02 Jun 2013 20:59:27 +0200

define-me (0.3.1.2-1ubuntu1) oneiric; urgency=low

  * Fixed a bug with preferences

 -- Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>  Sat, 15 Oct 2011 14:23:14 +0200

define-me (0.3.1.1-1ubuntu1) oneiric; urgency=low

  * Fixed some bugs with the search and with spaces

 -- Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>  Sat, 15 Oct 2011 14:09:49 +0200

define-me (0.3.1-1ubuntu1) oneiric; urgency=low

  * Fixed a bug with launcher

 -- Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>  Sat, 15 Oct 2011 07:54:19 +0200

define-me (0.3.0-1ubuntu1) oneiric; urgency=low

  * Ported to PyGi

 -- Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>  Wed, 12 Oct 2011 18:46:36 +0200

define-me (0.2.1-1ubuntu1) maverick; urgency=low

  * Fixed a bug

 -- Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>  Thu, 17 Feb 2011 21:18:47 +0100


src/preferences.py
#!/usr/bin/python
import gi
gi.require_version('Gtk', '3.0')

from gi.repository import Gtk
from gi.repository import GObject

import sys, re
import locale
import gettext
import comun
from configurator import Configuration

locale.setlocale(locale.LC_ALL, '')
gettext.bindtextdomain(comun.APP, comun.LANGDIR)
gettext.textdomain(comun.APP)
_ = gettext.gettext

languages = {}
languages[_('Deutch')]='de'
languages[_('English')]='en'
languages[_('French')]='fr'
languages[_('Spanish')]='es'
languages[_('Italian')]='it'

codes = {}
codes['de'] = _('Deutch')
codes['en'] = _('English')
codes['fr'] = _('French')
codes['es'] = _('Spanish')
codes['it'] = _('Italian')


urls = {}

urls['wikipedia'] = {'es':'http://es.wikipedia.org/wiki/',
					'en':'http://en.wikipedia.org/wiki/',
					'fr':'http://fr.wikipedia.org/wiki/',
					'it':'http://it.wikipedia.org/wiki/',
					'de':'http://de.wikipedia.org/wiki/'}
urls['wordreference'] = {'es':'http://www.wordreference.com/definicion/',
						'en':'http://www.wordreference.com/definition/',
						'fr':None,
						'it':'http://www.wordreference.com/definizione/',
						'de':None}
#urls['rae'] = {'es':'http://buscon.rae.es/draeI/SrvltGUIBusUsual?origen=RAE&LEMA='}
#urls['rae'] = {'es':'http://lema.rae.es/drae/?val='}
urls['rae'] = {'es':'http://buscon.rae.es/draeI/SrvltGUIBusUsual?LEMA=',
			   'en':None,
			   'fr':None,
			   'it':None,
			   'de':None}
#urls['rae'] = {'es':'http://lema.rae.es/drae/srv/search?val='}
#urls['rae'] = {'es':'http://buscon.rae.es/draeI/SrvltConsulta?TIPO_BUS=3&LEMA='}
urls['dictionary.com'] = {'es':'http://spanish.dictionary.com/definition/',
				'en':'http://dictionary.reference.com/browse/',
				'fr':None,
				'it':None,
				'de':None}
							  


class PF(Gtk.Dialog): # needs Gtk, Python, Webkit-Gtk
	def __init__(self,parent):
		Gtk.Dialog.__init__(self)
		title = comun.APPNAME + ' | '+_('Preferences')
		Gtk.Dialog.__init__(self, title,parent,Gtk.DialogFlags.MODAL | Gtk.DialogFlags.DESTROY_WITH_PARENT,(Gtk.STOCK_CANCEL, Gtk.ResponseType.REJECT,Gtk.STOCK_OK, Gtk.ResponseType.ACCEPT))
		self.set_position(Gtk.WindowPosition.CENTER_ALWAYS)
		self.connect('destroy', self.close_application)
		#
		frame = Gtk.Frame()
		self.get_content_area().add(frame)
		vbox1 = Gtk.VBox(spacing = 5)
		vbox1.set_border_width(5)
		frame.add(vbox1)
		#
		hbox1 = Gtk.HBox(spacing=5)
		vbox1.pack_start(hbox1,False,False,0)
		#
		label1 = Gtk.Label(_('Language')+': ')
		hbox1.pack_start(label1,False,False,0)
		#
		self.combobox1 = Gtk.ComboBox()
		self.model1 = Gtk.ListStore(str,str)
		for language in languages:
			self.model1.append([language,languages[language]])
		self.combobox1.set_model(self.model1)
		cell1 = Gtk.CellRendererText()
		self.combobox1.pack_start(cell1,True);
		self.combobox1.add_attribute(cell1,'text',0)
		self.combobox1.set_active(0)
		self.combobox1.connect('changed',self.on_selection_changed)
		hbox1.pack_start(self.combobox1,True,True,0)
		#
		hbox2 = Gtk.HBox(spacing=5)
		vbox1.pack_start(hbox2,False,False,0)
		#
		label2 = Gtk.Label(_('Prefered')+': ')
		hbox2.pack_start(label2,False,False,0)
		#
		self.combobox2 = Gtk.ComboBox()
		self.model2 = Gtk.ListStore(str,str)
		self.model2.append(['Wikipedia','wikipedia'])
		self.model2.append(['WordReference','wordreference'])
		self.model2.append(['RAE','rae'])
		self.model2.append(['Dictionary.com','dictionary.com'])
		self.combobox2.set_model(self.model2)
		cell2 = Gtk.CellRendererText()
		self.combobox2.pack_start(cell2,True);
		self.combobox2.add_attribute(cell2,'text',0)
		self.combobox2.set_active(0)
		hbox2.pack_start(self.combobox2,True,True,0)
		#
		self.load_preferences()
		#
		self.show_all()
		#
		if self.run() == Gtk.ResponseType.ACCEPT:
			self.save_preferences()
		self.destroy()
		
	
	def close_application(self, widget, data=None):
		self.hide()
	
	def on_selection_changed(self,widget):
		self.model2.clear()
		language_iter = self.combobox1.get_active_iter()
		language = self.model1.get_value(language_iter,1)
		if language in urls['wikipedia'].keys():
			self.model2.append(['Wikipedia','wikipedia'])
		if language in urls['wordreference'].keys():
			self.model2.append(['WordReference','wordreference'])
		if language in urls['rae'].keys():
			self.model2.append(['RAE','rae'])
		if language in urls['dictionary.com'].keys():
			self.model2.append(['Dictionary.com','dictionary.com'])
		self.combobox2.set_active(0)

	def load_preferences(self):
		configuration = Configuration()
		language = configuration.get('language')
		dictionary = configuration.get('dictionary')
		print(language)
		print(dictionary)
		itera = self.model1.get_iter_first()
		while itera != None:
			if self.model1.get_value(itera,1) == language:
				break					
			itera = self.model1.iter_next(itera)
		if itera != None:
			self.combobox1.set_active_iter(itera)
		else:
			self.combobox1.set_active(0)
		itera = self.model2.get_iter_first()
		while itera != None:
			if self.model2.get_value(itera,1) == dictionary:
				break					
			itera = self.model2.iter_next(itera)
		if itera != None:
			self.combobox2.set_active_iter(itera)
		else:
			self.combobox2.set_active(0)		
		
	def save_preferences(self):
		configuration = Configuration()
		language = self.model1.get_value(self.combobox1.get_active_iter(),1)
		dictionary = self.model2.get_value(self.combobox2.get_active_iter(),1)
		configuration.set('language',language)
		configuration.set('dictionary',dictionary)
		configuration.save()
		
if __name__ == '__main__':
	pf = PF(None)
	exit(0)



src/configurator.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
#
#
# Copyright (C) 2010 Lorenzo Carbonell
# lorenzo.carbonell.cerezo@gmail.com
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#
#
import codecs
import os
import json

import comun

class Configuration(object):
	def __init__(self):
		self.params = comun.PARAMS
		self.read()
	
	def get(self,key):
		try:
			return self.params[key]
		except KeyError as e:
			print(e)
			self.params[key] = comun.PARAMS[key]
			return self.params[key]
		
	def set(self,key,value):
		self.params[key] = value

	def reset(self):
		if os.path.exists(comun.CONFIG_FILE):
			os.remove(comun.CONFIG_FILE)		
		self.params = comun.PARAMS
		self.save()

	def set_defaults(self):
		self.params = comun.PARAMS
		self.save()
	
	def read(self):		
		try:
			f=codecs.open(comun.CONFIG_FILE,'r','utf-8')
		except IOError as e:
			print(e)
			self.save()
			f=codecs.open(comun.CONFIG_FILE,'r','utf-8')
		try:
			self.params = json.loads(f.read())
		except ValueError as e:
			print(e)
			self.save()
		f.close()

	def save(self):
		if not os.path.exists(comun.CONFIG_APP_DIR):
			os.makedirs(comun.CONFIG_APP_DIR)
		f=codecs.open(comun.CONFIG_FILE,'w','utf-8')
		f.write(json.dumps(self.params))
		f.close()


src/comun.py
#! /usr/bin/python
# -*- coding: iso-8859-1 -*-
#
__author__="atareao"
__date__ ="$09-jul-2011$"
#
# com.py
#
# Copyright (C) 2011 Lorenzo Carbonell
# lorenzo.carbonell.cerezo@gmail.com
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
#
#
__author__ = 'Lorenzo Carbonell <lorenzo.carbonell.cerezo@gmail.com>'
__date__ ='$09/07/2011'
__copyright__ = 'Copyright (c) 2011 Lorenzo Carbonell'
__license__ = 'GPLV3'
__url__ = 'http://www.atareao.es'
__version__ = '0.3.1.2'

import sys
import os
import locale
import gettext

######################################

def is_package():
    return __file__.find('src') < 0

######################################


APP = 'define-me'
APPNAME = 'Define-Me'
ICONNAME = 'define-me.svg'
PARAMS = {'language':'en',
		  'dictionary':'wikipedia',
		}

# check if running from source
if is_package():
    ROOTDIR = '/opt/extras.ubuntu.com/%s/share/'%APP
    LANGDIR = os.path.join(ROOTDIR, 'locale-langpack')
    APPDIR = os.path.join(ROOTDIR, APP)
    ICON = os.path.join('/opt/extras.ubuntu.com/%s/share/pixmaps'%APP,ICONNAME)
    CHANGELOG = os.path.join(APPDIR,'changelog')
else:
    ROOTDIR = os.path.dirname(__file__)
    LANGDIR = os.path.normpath(os.path.join(ROOTDIR, '../template1'))
    DATADIR = os.path.normpath(os.path.join(ROOTDIR, '../data'))
    ICON = os.path.join(DATADIR,ICONNAME)
    APPDIR = ROOTDIR
    DEBIANDIR = os.path.normpath(os.path.join(ROOTDIR, '../debian'))
    CHANGELOG = os.path.join(DEBIANDIR,'changelog')

APP_CONF = APP + '.conf'
CONFIG_DIR = os.path.join(os.path.expanduser('~'),'.config')
CONFIG_APP_DIR = os.path.join(CONFIG_DIR, APP)
CONFIG_FILE = os.path.join(CONFIG_APP_DIR, APP_CONF)

f = open(CHANGELOG,'r')
line = f.readline()
f.close()
pos=line.find('(')
posf=line.find('-',pos)
VERSION = line[pos+1:posf].strip()
if not is_package():
	VERSION = VERSION + '-src'

try:
	current_locale, encoding = locale.getdefaultlocale()
	language = gettext.translation(APP, LANGDIR, [current_locale])
	language.install()
	print(language)
	if sys.version_info[0] == 3:
		_ = language.gettext
	else:
		_ = language.ugettext
except Exception as e:
	print(e)
	_ = str
if sys.version_info[0] == 3:	
	import urllib.request
	def read_from_url(url):
		url = url.replace(' ','%20')
		request = urllib.request.Request(url, headers={'User-Agent' : 'Magic Browser'})
		f = urllib.request.urlopen(request)
		json_string = f.read()
		f.close()
		return json_string

	def internet_on():
		try:
			response=urllib.request.urlopen('http://google.com',timeout=1)
			return True
		except:
			pass
		return False
	def fromunicode(cadena):
		return cadena
else:
	import urllib2
	import urllib	
	def read_from_url(url):
		url = url.replace(' ','%20')
		headers = {
			'User-Agent' : 'Mozilla/4.0 (compatible; MSIE 5.5; Windows NT)',
			'Accept' : 'text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5',
			'Accept-Language' : 'es-es,fr-fr,en-us;q=0.7,en;q=0.3',
			'Accept-Charset' : 'ISO-8859-1,utf-8;q=0.7,*;q=0.7'
		}
		request = urllib2.Request(url, headers=headers)
		f = urllib2.urlopen(request)
		json_string = f.read()
		f.close()
		return json_string
		
	def internet_on():
		try:
			response=urllib2.urlopen('http://google.com',timeout=1)
			return True
		except:
			pass
		return False
	def fromunicode(cadena):
		if type(cadena) == unicode:
			ans = cadena.encode('utf-8')
			print(type(ans),ans)
			return ans
		return cadena
